app/controllers/api/v1/likes_controller.rb

```diff
@@ -0,0 +1,38 @@
+class Api::V1::LikesController < ApplicationController
+  protect_from_forgery unless: -> { request.format.json? }
+
+  def like_save(liked_record, params)
+    if liked_record.save
+      review = Review.find(params[:reviewId])
+      votes = review.votes_count
+      render json: { like: liked_record, votes: votes }
+    else
+      render json: { error: liked_record.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  def set_vote(params)
+    if params[:like_id]
+      liked_record = Like.find(params[:like_id])
+      if params[:liked] != params[:button].downcase
+        liked_record.vote = params[:button].downcase
+      else
+        liked_record.vote = nil
+      end
+      like_save(liked_record, params)
+    else
+      liked_record = Like.new(user_id: params[:currentUser], review_id: params[:reviewId], vote: params[:button].downcase)
+      like_save(liked_record, params)
+    end
+  end
+
```

Comment:
Good job extracting these to seperate messages. However, its a bit confusing in terms of placement and responsbility. `set_vote` and `like_save` should be `private` methods, defining at the bottom of this file. Make that change and I will accept this.

Reply:
Done!